addons/purchase/models/product.py

```diff
@@ -78,6 +78,41 @@ def action_view_po(self):
         }
         return action
 
+    @api.model
+    def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_uid=None):
+        if name and self.env.context.get('purchase_favorites'):
```

Comment:
Why avoid the computation if there is no name. you should remove the `if name`. 
The feature is very useful without typing anything (it means when name isn't set).